src/DiscoveryAnnouncer.coffee

```diff
@@ -1,83 +1,117 @@
-Promise        = require "bluebird"
-Errors         = require "./Errors"
+Promise = require "bluebird"
+Errors = require "./Errors"
 RequestPromise = require "./RequestPromise"
-Utils          = require "./Utils"
-_              = require "lodash"
+Utils = require "./Utils"
+_ = require "lodash"
+
+
+# this is now static because we have multiple discover annoucers per annouce region!
+gAnnouceAttempts = 20
+gInitialBackoff = 500
+
+resetGlobalCounts = () ->
+  gAnnouceAttempts = 20
+  gInitialBackoff = 500
 
 module.exports = class DiscoveryAnnouncer
 
-  constructor:(@logger, @serverList, @discoveryNotifier, @reconnect)->
-    @announcements = {}
-    @ANNOUNCE_ATTEMPTS = 20
-    @INITIAL_BACKOFF = 500
+  constructor: (@logger, @serverList, @discoveryNotifier, @announcementIndex, @announceRegion, @reconnect) ->
+    @announcedIds = []
 
-  pingAllAnnouncements:()=>
-    Promise.settle(_.map(@announcements, @attemptAnnounce))
+  pingAllAnnouncements: () =>
+    announcements = @getAnnouced()
+    Promise.settle(_.map(announcements, @attemptAnnounce) )
       .then(Utils.groupPromiseInspections)
-      .then (resultGroups)=>
+      .then (resultGroups) =>
         if resultGroups.rejected?.length > 0
           @logger.log "error", "#{resultGroups.rejected?.length} announcements failed"
         resultGroups.fulfilled
 
-  announce:(announcement, callback)->
+  getAnnouced: () =>
+    _.filter @announcementIndex.getAnnouncements(), (announcement) =>
+      announcement.announcementId in @announcedIds
+
+  announce: (announcement, callback) =>
     Utils.promiseRetry(
       @attemptAnnounce.bind(@, announcement)
-      @ANNOUNCE_ATTEMPTS
-      @INITIAL_BACKOFF
+      gAnnouceAttempts
+      gInitialBackoff
     ).nodeify(callback)
 
-  removeAnnouncement:(announcement)->
-    delete @announcements[announcement.announcementId]
+  resetRetryCounts: () ->
+    resetGlobalCounts()
+
+  setAnnouceAttemptsCount: (attemptsCount) ->
+    gAnnouceAttempts = attemptsCount
+
+  setInitialBackoffCount: (backoffCount) ->
+    gInitialBackoff = backoffCount
```

Comment:
for eg, we wouldn't need all of these three to reset the try counts for testing.